include/wx/defs.h

```diff
@@ -279,6 +279,17 @@ typedef short int WXTYPE;
     #define wxNODISCARD
 #endif
 
+/* wxWARN_UNUSED is used as an attribute to a class, stating that unused instances
+   should be warned about (in case such warnings are enabled in the first place) */
+
+#ifdef __has_attribute
```

Comment:
Why do we use `__has_attribute` here and not `__has_cpp_attribute` that we use everywhere else? Is there really some difference between them (in C++ code, that is)? If not, I'd prefer to use the latter for consistency.

Reply:
I suggested `__has_attribute` because `include/wx/defs.h` is a C file, and `__has_cpp_attribute` is not available in C mode with Clang.

Reply:
Understood, but I don't think we care about using this in C code (we are not going to do it ourselves and we won't be adding it to 3rd party C code we use), so using C++ check still seems better to me.

Thanks for updating this, Lauri, I'll merge it soon.